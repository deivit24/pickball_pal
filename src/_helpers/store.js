import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
