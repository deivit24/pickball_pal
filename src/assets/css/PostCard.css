.post-card-box {
  position: relative;
  border-radius: 25px;
  box-shadow: 0 0 4px 3px #afafaf;
  padding: 10px 20px;
}

.post-card-box .panel-footer span:hover {
  cursor: pointer;
}
.post-card-box .panel-footer span:active {
  color: green;
}
.post-card-box .panel-heading {
  margin-top: 20px;
  padding-bottom: 5px;
  display: flex;
}
.post-card-box .panel-heading img {
  margin-right: 15px;
  width: 80px;
  border-radius: 50%;
}
.post-card-box .panel-heading h3 {
  margin: 0px;
  height: 80px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  font-weight: 700;
}
.post-card-box .panel-heading h5 {
  color: rgb(153, 153, 153);
  margin: 0px;
  height: 80px;
  font-size: 15px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  font-weight: 700;
  margin-left: auto;
}
.post-card-box .panel-body {
  padding-top: 5px;
  font-size: 13px;
}
.post-card-box .panel-body .post-card-box-image {
  display: block;
  text-align: center;
  background-color: rgb(245, 245, 245);
  border: 1px solid rgb(217, 217, 217);
}
.post-card-box .panel-body .post-card-box-image img {
  max-width: 100%;
}

.post-card-box .panel-footer {
  font-size: 14px;
  font-weight: 700;
  min-height: 54px;
  display: flex;
  justify-content: center;
  align-items: center;
}
.post-card-box .panel-footer .btn {
  float: left;
  margin-right: 8px;
}
.post-card-box .panel-footer .input-placeholder {
  width: 90%;
  margin-left: auto;
  color: rgb(153, 153, 153);
  font-size: 12px;
  font-weight: 400;
  padding: 8px 6px 7px;
  border: 1px solid rgb(217, 217, 217);
  border-radius: 2px;
  box-shadow: rgba(0, 0, 0, 0.0470588) 0px 1px 0px 0px;
}
.post-card-box.post-card-box-show-comment .panel-footer .input-placeholder {
  display: none;
}
.post-card-box .post-card-box-comment {
  display: none;
  padding: 10px 20px 15px;
  border-top: 1px solid rgb(229, 229, 229);
  background-color: rgb(245, 245, 245);
}
.post-card-box.post-card-box-show-comment .post-card-box-comment {
  display: block;
}
/*.post-card-box .post-card-box-comment img {
  float: left;   
}*/
.post-card-box .post-card-box-comment .post-card-box-textarea {
  float: right;
  width: calc(100% - 56px);
}
.post-card-box .post-card-box-comment .post-card-box-textarea textarea {
  display: block;
  /*margin-left: 60px;
  width: calc(100% - 56px);*/
  width: 100%;
  background-color: rgb(255, 255, 255);
  border: 1px solid rgb(217, 217, 217);
  box-shadow: rgba(0, 0, 0, 0.0470588) 0px 1px 0px 0px;
  resize: vertical;
}
.post-card-box .post-card-box-comment .post-card-box-textarea .btn {
  margin-top: 10px;
  margin-right: 8px;
  width: 100%;
}

#DeleteButton {
  position: absolute;
  top: 10px;
  right: 0;
}
@media (min-width: 992px) {
  .post-card-box .post-card-box-comment .post-card-box-textarea .btn {
    width: auto;
  }
}

.post-card-box .btn {
  border-radius: 3px;
}
.post-card-box .btn-default {
  border: 1px solid rgb(217, 217, 217);
  box-shadow: rgba(0, 0, 0, 0.0470588) 0px 1px 0px 0px;
}
.post-card-box .btn-default:hover,
.post-card-box .btn-default:focus,
.post-card-box .btn-default:active {
  background-color: rgb(255, 255, 255);
  border-color: rgb(0, 0, 0);
}
