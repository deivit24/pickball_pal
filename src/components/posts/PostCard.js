// React
import React from 'react';
// Redux
import { useSelector, useDispatch } from 'react-redux';
// React Router DOM
import { Link } from 'react-router-dom';
// Actions from Redux
import { addLike, removeLike, deletePost } from '../../_actions/post';
// Comment Form Component
import CommentForm from '../post/CommentForm';
// Moment
import Moment from 'react-moment';
import { Icon, Badge, Button } from '@material-ui/core';
// Post Card CSS
import '../../assets/css/PostCard.css';

const PostCard = ({
  showActions,
  post: {
    _id,
    text,
    first_name,
    last_name,
    avatar,
    user,
    location,
    likes,
    comments,
    date,
  },
}) => {
  const dispatch = useDispatch();

  let name =
    first_name.charAt(0).toUpperCase() +
    first_name.slice(1) +
    ' ' +
    last_name.charAt(0).toUpperCase() +
    last_name.slice(1);

  const auth = useSelector((store) => store.auth);

  return (
    <div id="PostCard" className="row mb-3">
      <div className="col-12 col-sm-offset-1 ">
        <div className=" post-card-box">
          <div className="panel-heading">
            <Link to={`/profile/${user}`}>
              <img
                className="[ img-circle pull-left ]"
                src={avatar}
                alt="Mouse0270"
              />
            </Link>
            <Link to={`/profile/${user}`}>
              <h3>{name}</h3>
              <p>{location}</p>
            </Link>

            <h5>
              Posted
              <Moment to={date} />
            </h5>
          </div>
          <div className="panel-body">
            <p>{text}</p>
          </div>

          {showActions ?? (
            <>
              <div className="panel-footer">
                <Badge
                  onClick={(e) => dispatch(addLike(_id))}
                  className="mr-3"
                  anchorOrigin={{
                    vertical: 'bottom',
                    horizontal: 'left',
                  }}
                  badgeContent={likes.length}
                  color="primary"
                >
                  <Icon className="fas fa-thumbs-up" />
                </Badge>

                <Badge
                  onClick={(e) => dispatch(removeLike(_id))}
                  anchorOrigin={{
                    vertical: 'bottom',
                    horizontal: 'left',
                  }}
                  className="mr-3"
                  color="primary"
                >
                  <Icon className="fas fa-thumbs-down" />
                </Badge>
                <Link to={`/posts/${_id}`}>
                  <Badge
                    anchorOrigin={{
                      vertical: 'bottom',
                      horizontal: 'left',
                    }}
                    className="mr-3"
                    badgeContent={comments.length}
                    color="primary"
                  >
                    <Icon className="fas fa-comments" />
                  </Badge>
                </Link>
                {!auth.loading && user === auth.user._id && (
                  <Button
                    id="DeleteButton"
                    onClick={(e) => dispatch(deletePost(_id))}
                    color="secondary"
                  >
                    <Icon className="fas fa-trash"></Icon>
                  </Button>
                )}
                <CommentForm postId={_id} color="black" />
              </div>
            </>
          )}
        </div>
      </div>
    </div>
  );
};

export default PostCard;
