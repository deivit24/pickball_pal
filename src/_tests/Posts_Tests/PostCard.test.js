import React from 'react';
import { render } from '@testing-library/react';
import PostCard from '../../components/posts/PostCard';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing PostCard

describe('Testing the PostCard', () => {
  let showActions = true;
  let post = {
    _id: '00002',
    text: 'hey this is a comment',
    first_name: 'david',
    last_name: 'salazar',
    avatar:
      'https://icon-library.com/images/default-user-icon/default-user-icon-4.jpg',
    user: '00003',
    date: Date.now(),
    location: 'Austin, TX 78745',
    likes: 2,
    comments: 3,
  };
  it('it renders PostCard without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <PostCard showActions={showActions} post={post} />
        </Provider>
      </MemoryRouter>
    );
  });

  // Wont work because there is a Date.now() and it changes from snapshot
  // it('matches snapshot', () => {
  //   const { asFragment } = render(
  //     <MemoryRouter>
  //       <Provider store={store}>
  //         <PostCard showActions={showActions} post={post} />
  //       </Provider>
  //     </MemoryRouter>
  //   );
  //   expect(asFragment()).toMatchSnapshot();
  // });
});
