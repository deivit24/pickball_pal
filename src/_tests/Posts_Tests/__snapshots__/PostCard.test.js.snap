// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the PostCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row mb-3"
    id="PostCard"
  >
    <div
      class="col-12 col-sm-offset-1 "
    >
      <div
        class=" post-card-box"
      >
        <div
          class="panel-heading"
        >
          <a
            href="/profile/00003"
          >
            <img
              alt="Mouse0270"
              class="[ img-circle pull-left ]"
              src="https://icon-library.com/images/default-user-icon/default-user-icon-4.jpg"
            />
          </a>
          <a
            href="/profile/00003"
          >
            <h3>
              David Salazar
            </h3>
            <p>
              Austin, TX 78745
            </p>
          </a>
          <h5>
            Posted
            <time
              datetime="1601072880285"
            >
              a few seconds ago
            </time>
          </h5>
        </div>
        <div
          class="panel-body"
        >
          <p>
            hey this is a comment
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
